AEPAudience/Sources/AudienceState.swift

```diff
@@ -88,19 +88,20 @@ class AudienceState {
         }
 
         // if the event is a lifecycle event, convert the lifecycle keys to audience manager keys
-        if event.type == EventType.lifecycle {
+        var signalData: [String: String] = [:]
+        if event.type == EventType.lifecycle, let eventData = event.data, !eventData.isEmpty {
```

Comment:
nit: as `!eventData.isEmpty` is a condition for both `if` and `else`, can you move that to an outer `if` block?
```
if let eventData = event.data, !eventData.isEmpty {
   if event.type == EventType.lifecycle {
       // do lifecycle stuff
   } else {
      // do signal traits stuff
   }
}
```